chrome/browser/android/adblock/adblock_bridge.cc

```diff
@@ -119,46 +119,36 @@ std::string escapeJavaScriptString(const std::string& subject) {
     U2029, "\\u2029");
 }
 
-std::string generateJavascript(
+const uint SELECTOR_GROUP_SIZE = 1024;
+#define DISPLAY_NONE_IMPORTANT " {display: none !important;}\n"
+
+std::string generateStylesheet(
   AdblockPlus::FilterEnginePtr filterEngine,
   std::string url,
   std::string domain) {
-  LOG(WARNING) << "Adblock: getting selectors for domain " << domain;
+  std::stringstream stylesheet;
   std::vector<std::string> selectors = filterEngine->GetElementHidingSelectors(domain);
-  LOG(WARNING) << "Adblock: got " << selectors.size() << " selectors for domain " << domain;
-
-  // build the string with selectors
-  std::stringstream ss;
-  ss << "var selectors = [\n";
-  for (unsigned int i=0; i<selectors.size(); i++) {
-    if (i > 0) {
-      ss << ", ";
+  
+  // selectors list should be splitted into groups of SELECTOR_GROUP_SIZE size:
+  // https://github.com/adblockplus/adblockpluschrome/blob/33991b8d88dae2712a9a3030d5143badbc155718/lib/contentFiltering.js#L46
+  
+  unsigned int last_group = 0;
+  for (unsigned int i = 0; i < selectors.size(); i++) {
+    unsigned int group = i / SELECTOR_GROUP_SIZE;
+    if (group > last_group) {
+      stylesheet << DISPLAY_NONE_IMPORTANT;
+    } else {
+      if (i > 0) {
+        stylesheet << ", ";
+      }
     }
-    //LOG(WARNING) << "Adblock: selector: " << selectors[i];
-    ss << "\"" << escapeJavaScriptString(selectors[i]) << "\"\n";
+    last_group = group;
+    stylesheet << selectors[i];
+  }
+  if (selectors.size() > 0) {
```

Comment:
The condition should check whether there are still some selectors without the rule.

Reply:
> are still some selectors without the rule
what do you mean? we've iterated all the list

Reply:
> are still some selectors without the rule
what do you mean? we've iterated the list